src/Form/DataTransformerResolver.php

```diff
@@ -108,8 +108,8 @@ private function hasFieldDescriptionAssociationWithClass(
         FieldDescriptionInterface $fieldDescription,
         string $class
     ): bool {
-        return (
-            method_exists($fieldDescription, 'getTargetModel') && $class === $fieldDescription->getTargetModel()
-        ) || $class === $fieldDescription->getTargetEntity();
+        return method_exists($fieldDescription, 'getTargetModel') ?
```

Comment:
I found this while changing all the tests, this can be done on a separate PR if you want.

Reply:
This PR could take some times before being merged. So maybe it's indeed a good idea to create a separate PR

Reply:
Let's see how this goes, if we see a lot of review process I will revert it from this PR and open a new one. If not, I will change it to patch and add a changelog

Reply:
Yes, but at least a separate commit seems needed.